### PR TITLE
[Driver][SYCL] Update integration footer behavior to not do preprocess step

### PR DESCRIPTION
**Pull request by mdtoguchi**
When performing integration footer work, we would preprocess the host
    compilation source before appending the footer.  Adjust this behavior to
    not preprocess and just append.

    The following occurs:
      - # line directive is inserted at the front of the file to enable
        proper source file diagnostic information
      - integration footer is not appended, but rather #included

### CODE REVIEW

**Review by AGindinson — DISMISSED** [review]
LGTM; especially liking the #line directive trick!

**Review by AGindinson — APPROVED** [review]

**Review by AGindinson — DISMISSED** [review]

**Comment by bader** [comment]
@mdtoguchi, it looks like there are regressions in Jeknins/Summary tests. Could you take a look, please?

File: llvm/tools/append-file/append-file.cpp [hunk 1]
```diff
@@ -52,17 +63,45 @@ int main(int argc, const char **argv) {
...
-  // Copy the input file to the output file
-  llvm::sys::fs::copy_file(Input, Output);
+  // Open the original source file stream.
+  std::ifstream InputFile(Input, std::ios_base::binary);
+  // Open the output file stream.
+  std::ofstream OutFile(Output, std::ios_base::binary);
+
+  // Add the BOM from the original source (if it exists).  We will handle
+  // UTF-8, UTF-16LE and UTF-16BE
+  unsigned char char1 = InputFile.get();
```
Comment (AGindinson) [inline llvm/tools/append-file/append-file.cpp:73]: (Minor) suggestion of the guideline-conforming renaming
```suggestion
  unsigned char Char1 = InputFile.get();
```

**Review by AGindinson — APPROVED** [review]
The logic of the BOM changes LGTM.

**Review by AGindinson — DISMISSED** [review]

**Comment by bader** [comment]
@mdtoguchi, please, take a look at regressions in Jenkins/Summary and resolve merge conflicts in `clang/test/Driver/sycl-intelfpga-aoco.cpp`.

**Review by AGindinson — APPROVED** [review]
Latest updates LGTM.

**Comment by AGindinson** [comment]
Pre-commit failures are not related to the patch - similar stuff happening at http://icl-jenkins.sc.intel.com:8080/blue/organizations/jenkins/LLVM-Test-Suite-CI-TMP%2FLLVM-Test-Suite-CI-WIN/detail/LLVM-Test-Suite-CI-WIN/321/pipeline/

**Comment by AlexeySachkov** [comment]
> Pre-commit failures are not related to the patch - similar stuff happening at http://icl-jenkins.sc.intel.com:8080/blue/organizations/jenkins/LLVM-Test-Suite-CI-TMP%2FLLVM-Test-Suite-CI-WIN/detail/LLVM-Test-Suite-CI-WIN/321/pipeline/

@AGindinson, do we have a tracker for those issues? I suggest that we also go ahead and temporary disable those tests until we fixed them

**Comment by AGindinson** [comment]
> I suggest that we also go ahead and temporary disable those tests until we fixed them

This is being tracked, so I share the viewpoint that we should merge this patch despite the failures once the summary tests are over.

**Comment by mdtoguchi** [comment]
> > I suggest that we also go ahead and temporary disable those tests until we fixed them
> 
> This is being tracked, so I share the viewpoint that we should merge this patch despite the failures once the summary tests are over.

Where is this being tracked?  I'd like to add a reference here so it isn't lost.

**Pull request merged**